Dockerfile

```diff
@@ -8,7 +8,7 @@ COPY cmd/    cmd/
 COPY vendor/ vendor/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager sigs.k8s.io/cluster-api-provider-gcp/cmd/manager
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -ldflags '-extldflags "-static"' -o manager sigs.k8s.io/cluster-api-provider-gcp/cmd/manager
```

Comment:
It is weird that we do this, but ... I agree.  The `cc` distroless image does include libc and thus avoids the need for static linking, but this feels better.  I think there's some limitations around static linking on osx, but we're in docker anyway...